docs/types.rst

```diff
@@ -608,6 +607,24 @@ number of bytes, always use one of ``bytes1`` to ``bytes32`` because they are mu
     that you are accessing the low-level bytes of the UTF-8 representation,
     and not the individual characters!
 
+You can initialize an array inline using syntax such as ``string[] myarray = ["a", "b"];``. However, this only works for statically sized memory arrays. For example.
+
+::
+
+    pragma solidity ^0.4.16;
+
+    contract C {
+        function f() public pure returns (uint8[5]) {
+            string[4] memory adaArr = ["This", "is", "an", "array"];
+            return ([1, 2, 3, 4, 5]);
+        }
+    }
+
+You can create and initialize multi-dimensional arrays in the same ways, but note that filling arrays can cost a lot of gas, so make sure you optimize how you use them.
+
+...note::
+    Optimizing storage access can reduce the gas costs considerably, because you can store 32 ``uint8`` values in a single slot. These optimizations do not work across loops and also have a problem with bounds checking.
```

Comment:
`have a problem with bounds checking` - we should either explain the problem or not mention it at all.

Reply:
@chriseth 

I actually feel that this whole note:

> ...note::
    Optimising storage access can reduce the gas costs considerably, because you can store 32 ``uint8`` values in a single slot. These optimisations do not work across loops and also have a problem with bounds checking.


Is kind of important and should be mentioned more explicitly elsewhere if it isn't already.

Reply:
Yes, we can do that, but in the end, the optimizer is not working that well and using smaller types sometimes increases gas costs, so I'm not sure what I would recommend.

Reply:
@chriseth OK, I'm going to stash this idea for now for future discussion.